homeassistant/bootstrap.py

```diff
@@ -507,6 +507,17 @@ async def _async_set_up_integrations(
 
     _LOGGER.info("Domains to be set up: %s", domains_to_setup)
 
+    # Load the registries
+    await asyncio.gather(
+        device_registry.async_load(hass),
+        entity_registry.async_load(hass),
+        area_registry.async_load(hass),
+    )
+
+    # Initialize recorder
+    if "recorder" in domains_to_setup:
+        recorder.async_initialize_recorder(hass)
```

Comment:
This adds an `asyncio.Future` to `hass.data` which integrations (frontend) can wait on until recorder has concluded if a non-live migration is needed or not.